### PR TITLE
Fix crash during TaskScheduler creation

### PR DESCRIPTION
**Pull request by sobkas**
Fixes run when compiled with gcc higher than 10

Tested it on gcc-14 and binary worked.
But need more tests.

### CODE REVIEW

**Comment by sobkas** [comment]
Ok, fixed branch

**Comment by Xottab-DUTY** [comment]
~~What's the issue?~~

image

File: src/xrCore/Threading/TaskManager.hpp [hunk 1]
```diff
@@ -78,6 +78,7 @@ class XRCORE_API TaskManager final
 public:
     void RegisterThisThreadAsWorker();
     void UnregisterThisThreadAsWorker();
+    void MakeThreads();
```
Comment (Xottab-DUTY) [inline src/xrCore/Threading/TaskManager.hpp]:
```suggestion
    void SpawnThreads();
```

File: src/xrCore/xrCore.cpp [hunk 2]
```diff
@@ -274,6 +274,7 @@ void xrCore::Initialize(pcstr _ApplicationName, pcstr commandLine, bool init_fs,
         Msg("\ncommand line %s\n", Params);
         _initialize_cpu();
         TaskScheduler = xr_make_unique<TaskManager>();
+        TaskScheduler->MakeThreads();
```
Comment (Xottab-DUTY) [inline src/xrCore/xrCore.cpp]:
```suggestion
        TaskScheduler->SpawnThreads();
```

File: src/xrCore/Threading/TaskManager.cpp [hunk 3]
```diff
@@ -230,6 +219,20 @@ TaskManager::~TaskManager()
     }
 }
 
+void TaskManager::MakeThreads()
```
Comment (Xottab-DUTY) [inline src/xrCore/Threading/TaskManager.cpp]:
```suggestion
void TaskManager::SpawnThreads()
```

**Review by Xottab-DUTY — APPROVED** [review]

**Comment by sobkas** [comment]
As it fixes problems with gcc newer than 10, maybe adding that to build information should be ok?

**Pull request merged**